src/Validation/Constraint/SignedWith.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolationException;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 4.0.0
+ */
+final class SignedWith implements Constraint
+{
+    /**
+     * @var Signer
+     */
+    private $signer;
+
+    /**
+     * @var Signer\Key
+     */
+    private $key;
+
+    public function __construct(Signer $signer, Signer\Key $key)
+    {
+        $this->signer = $signer;
+        $this->key = $key;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function assert(Token $token)
+    {
+        $signature = $token->signature();
+
+        if ($signature === null) {
+            throw new ConstraintViolationException('The token is not signed');
+        }
+
+        if ($token->headers()->get('alg', 'invalid') !== $this->signer->getAlgorithmId()) {
```

Comment:
What guarantees that `invalid` cannot be used by a custom algo? Why not use `null`?